node/src/components/block_proposer.rs

```diff
@@ -326,49 +271,46 @@ impl BlockProposer {
         ttl_valid && timestamp_valid && deploy_valid && num_deps_valid && all_deps_resolved()
     }
 
-    /// Notifies the block proposer of a new block that has been proposed, so that the block's
-    /// deploys are not returned again by `remaining_deploys`.
-    fn added_block<I>(&mut self, block: ProtoBlockHash, deploys: I)
+    /// Notifies the block proposer that a block has been finalized.
+    fn finalized_block<I>(&mut self, deploys: I)
```

Comment:
The only argument of this method is `deploys: Iter<Item = DeployHash>`, we should probably rename it and fix the comment.

Reply:
It is still called when a block is finalized, though. I would just explain in the comment that the argument is the list of deploys contained within the block that was finalized, if that sounds okay.

Reply:
There's no notification or block involved. I think it could be called `finalized_deploys`. It's a local method.